### PR TITLE
feat(workflow): Creating a model to store alert rule activity

### PR DESCRIPTION
**Pull request by iProgramStuff**
As part of bringing the alert rules into the alert stream, we're going to show some additional data in the rule listing. In particular, we now want to show who created the rule.

This could have been a field on the AlertRule model, but the hope here is that in the future we can track more alert rule actions (like updating, deleting, disabling, enabling (whether automatic due to downgrade or manual), etc...). So, much like we're doing for incidents, we've created a model that can be used to store all of the information around this.

### CODE REVIEW

File: src/sentry/incidents/models.py [hunk 1]
```diff
@@ -623,6 +623,30 @@ def get_registered_types(cls):
...
+    UPDATED = 3
+    ENABLED = 4
+    DISABLED = 5
+
+
+class AlertRuleActivity(Model):
+    __core__ = True
+
+    alert_rule = FlexibleForeignKey("sentry.AlertRule")
+    previous_alert_rule = FlexibleForeignKey(
```
Comment (markstory) [inline src/sentry/incidents/models.py:638]: When does this column get used?

**Review by markstory — COMMENTED** [review]
Schema setup looks good to me.

File: src/sentry/incidents/models.py (hunk 1 above)
Comment (iProgramStuff) [reply src/sentry/incidents/models.py:638]: Whenever an alert rule is changed, and it has associated incidents/alerts, we create a new instance of the rule and archive the old one (the old one remains the same, and the newly created one reflects their most recent edits). In those cases, I thought storing a reference to the rule it "came from" could be helpful down the line.

Comment (markstory) [reply src/sentry/incidents/models.py:638]: Makes sense to me.

**Review by markstory — APPROVED** [review]

**Review by wedamija — APPROVED** [review]
Looks good to me. I was wondering if it made sense to put this activity into the audit log, but I think there's a good case to have a separate table here.

**Pull request merged**